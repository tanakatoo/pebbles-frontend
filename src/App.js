import './styles/app.css';
import NavBar from "./components/common/NavBar"
import "./styles/app.css"
import { useSelector } from 'react-redux';
import MyRoutes from "./MyRoutes"

function App() {
  //get the font from the store
  const font = useSelector(state => state.langFont.font)

  //we have to hardcode the string of the font because tailwind doesn't know what to use at build
  return (
    <div className={`App ${font == 'font-poppins' ? 'font-poppins' : 'font-notoJP'}`}>
      <NavBar />
      <MyRoutes />
    </div >
  );
}

export default App;
