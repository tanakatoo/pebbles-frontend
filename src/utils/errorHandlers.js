import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}



export const errorHandlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {
        const failLogin = {
            "error": {
                "message": "INVALID_CREDENTIALS",
                "status": 401
            }
        }
        console.log(res(ctx.json([failLogin])))
        return res(ctx.json([failLogin]));
    }),

    rest.post('http://localhost:3001/email', (req, res, ctx) => {
        console.log('MOCK error /email');
        const noData = {
            "error": {
                "message": "no data",
                "status": 400
            }
        }
        return res(ctx.json(noData));
    }),

]