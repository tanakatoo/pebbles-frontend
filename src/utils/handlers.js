import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',

    }
}


export const handlers = [
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {


        console.log(res(ctx.json(loginResponse)))
        return res(
            // Respond with a 200 status code
            // ctx.status(200),
            res(ctx.json(loginResponse))
        )
    }),

    // Called via <ExploreCommunity type="studyBuddy" />
    rest.get('http://localhost:3001/study-buddies/1', (req, res, ctx) => {
        console.log('MOCK /study-buddies/1');
        console.log('  response:', {});
        return res(ctx.json({}));
    })
]