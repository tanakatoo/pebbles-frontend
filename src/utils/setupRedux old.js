
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from "../reducers/rootReducer"
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';

import storage from 'redux-persist/lib/storage'
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

export const persistConfig = {
    key: 'root', // Key prefix for the persisted state
    storage, // Storage mechanism (e.g., local storage)
    stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
    whitelist: ['profile', 'langFont']
};

export const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
            }
        })
})
export const persistedStore = persistStore(store)