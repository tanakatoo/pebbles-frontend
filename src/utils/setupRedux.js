import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react'
import { ListSlice } from "../reducers/rootReducer";
import rootReducer from "../reducers/rootReducer"
import storage from 'redux-persist/lib/storage'
import storeSetup from "./storeSetup"
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

// const persistConfig = {
//     key: 'root', // Key prefix for the persisted state
//     storage, // Storage mechanism (e.g., local storage)
//     stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
//     whitelist: ['profile', 'langFont']
// };
// const persistedReducer = persistReducer(persistConfig, rootReducer);

// const persistedStore = persistStore(store)
// const root = ReactDOM.createRoot(document.getElementById('root'));


export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = storeSetup(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {

        return <Provider store={store}>{children}</Provider>;
    }

    // Return an object with the store and all of RTL's query functions
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}