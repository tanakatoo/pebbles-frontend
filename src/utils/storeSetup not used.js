import { combineReducers, configureStore } from '@reduxjs/toolkit'

import rootReducer from '../reducers/rootReducer'
// import userReducer from '../features/users/userSlice'

// Create the root reducer separately so we can extract the RootState type
// const rootReducer = combineReducers({
//     user: userReducer
// })

export const setupStore = preloadedState => {
    return configureStore({
        reducer: rootReducer,
        preloadedState
    })
}