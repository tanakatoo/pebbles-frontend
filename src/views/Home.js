import React from "react"
import usePageText from "../hooks/usePageText"
import SearchBar from "../components/common/SearchBar"
import { Connect, Language, Travel } from "../styles/Icons"
import Mission from "../components/home/Mission"
import Supports from "../components/home/Supports"
import studybuddysquare from "../images/studyBuddy-square.jpg"
import communitysquare from "../images/community-square.jpg"
import regionalsquare from "../images/regional-square.jpg"
import marketplacesquare from "../images/marketplace-square.jpg"
import studysupportsquare from "../images/studysupport-square.jpg"
import { useSelector } from "react-redux"
import { Button } from "../components/button/Button"

const Home = () => {
    const pageText = usePageText("home")
    const lang = useSelector(state => state.langFont.lang)

    return (
        <div>
            <div className="bg-primary-dark py-6 px-4">
                <SearchBar />
            </div>
            <div className='flex justify-center bg-primary-super-light bg-homeHero md:bg-homeHeroTablet lg:bg-homeHeroDesktop bg-cover bg-center bg-no-repeat relative min-h-screen' >
                <div className="bg-gradient-to-b absolute from-primary-super-light from-0% via-transparent via-1% to-transparent to-100% w-full h-screen"></div>
                <h1 className="text-primary-dark mx-4 text-center pt-10 md:pt-16 text-mobile-header-1-homepage absolute">{pageText.H1}</h1>
            </div>
            <section className="pt-12 md:pt-24  mx-auto">
                <h2 className="text-center text-mobile-header-2-homepage text-primary-dark">{pageText.H2}</h2>
                <div className="mt-8 md:mt-12 lg:ps-8 lg:pe-8 flex flex-col justify-center items-center gap-8 lg:flex-row lg:w-full lg:space-around">
                    <div className="mb-5 flex flex-col lg:w-full">
                        <Mission icon={<Connect />} text={pageText.MISSION1} />
                    </div>
                    <div className="mb-5  flex flex-col lg:w-full">
                        <Mission icon={<Travel />} text={pageText.MISSION2} />
                    </div>
                    <div className="mb-5  flex flex-col lg:w-full">
                        <Mission icon={<Language />} text={pageText.MISSION3} />
                    </div>
                </div>
            </section >
            <section className="pt-[84px]">
                <div className=" flex relative bg-homeHelp md:bg-homeHelpTablet lg:bg-homeHelpDesktop h-[216px] md:h-[300px] lg:h-[350px] bg-cover bg-center">
                    <div className="bg-gradient-to-b from-white absolute w-full h-[216px] z-10 "></div>
                    <div className="w-[80%] md:w-[60%] lg:w-[50%] top-[180px] md:top-[280px] lg:top-[330px] left-1/2  -translate-x-1/2 text-center py-6 px-8 absolute bg-secondary text-white rounded-ml"><p>{pageText.HERO2}</p></div>
                </div>
            </section>
            <section className="pt-[120px]">
                <h2 className="text-mobile-header-2-homepage text-center text-primary-dark">{pageText.HOW_TO_HELP}</h2>
                <div className="mt-16">
                    {lang === "JA" ?
                        <Supports
                            font={`NotoSansJPMedium`}
                            bgColor='bg-background'
                            img={studysupportsquare}
                            title={pageText.HELP5_TITLE}
                            desc={pageText.HELP5_DESC} />
                        : ''
                    }
                    <Supports
                        font={`${lang === "EN" ? 'font-StudyBuddyEN' : 'font-StudyBuddyJA'}`}
                        bgColor='bg-study-buddy-accent'
                        img={studybuddysquare}
                        title={pageText.HELP3_TITLE}
                        desc={pageText.HELP3_DESC} />
                    <Supports
                        font={`font-Community`}
                        img={communitysquare}
                        title={pageText.HELP1_TITLE}
                        desc={pageText.HELP1_DESC}
                        bgColor={'bg-community-accent'}
                        titleColor="text-white"
                        descColor='text-white'
                        button=
                        {<Button btnText="Tell me more" textColor="text-primary-dark" py="py-2" bkColor="bg-white" type="button" />} />
                    <Supports
                        font={`font-Regional`}
                        img={regionalsquare}
                        title={pageText.HELP2_TITLE}
                        desc={pageText.HELP2_DESC}
                        bgColor={'bg-regional-accent'}
                        titleColor="text-white"
                        descColor='text-white'
                        button=
                        {<Button btnText="Tell me more" textColor="text-primary-dark" py="py-2" bkColor="bg-white" type="button" />} />
                    <Supports
                        font={`${lang === "EN" ? 'font-EnglishMarketEN' : 'font-EnglishMarketJA'}`}
                        bgColor='bg-marketplace-accent'
                        img={marketplacesquare}
                        title={pageText.HELP4_TITLE}
                        desc={pageText.HELP4_DESC} />


                </div>
            </section>
            <section className="py-12">
                <p className="py-8 px-4 text-mobile-page-header text-center text-primary-dark">{pageText.HERO3_TOP}</p>
                {/* <div className="flex relative bg-homeStartMobile md:bg-homeStartTablet lg:bg-homeStartDesktop h-[216px] md:h-[300px] lg:h-[350px] bg-cover bg-center">
                    <div className="bg-gradient-to-b absolute w-full h-[216px] z-10 from-white"></div>
                </div> */}
                <p className="px-4 text-mobile-page-header text-center text-primary-dark">{pageText.HERO3_BOTTOM}</p>
                <div className="flex justify-center">
                    <Button btnText="More about us" extraClasses="mt-12 w-[200px]"></Button>
                </div>
            </section>
        </div >
    )

}

export default Home