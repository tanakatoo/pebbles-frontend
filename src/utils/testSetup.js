// import { persistedStore, store } from './setupRedux';
import { render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { setupTestStore } from './setupRedux';


const renderWithProviders = (ui, options = {}) => {
    const [store, persistedStore] = setupTestStore(options.preloadedState);

    const AllProviders = ({ children }) => {
        return (
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistedStore}>
                    {children}
                </PersistGate>
            </Provider>
        )
    };

    return { store, ...render(ui, { wrapper: AllProviders, ...options }) };
}


// "re-export everything"
export * from '@testing-library/react';

export { renderWithProviders };