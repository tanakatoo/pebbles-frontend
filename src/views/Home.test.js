import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithProviders } from '../utils/testSetup';
import Home from './Home';
import App from '../App';


// Mock window.scrollTo()
window.scrollTo = jest.fn();
afterAll(() => {
    jest.clearAllMocks();
});


describe('Home', () => {
    test('renders without crashing', () => {
        renderWithProviders(
            <MemoryRouter initialEntries={["/"]}>
                <App />
            </MemoryRouter>
        );
    });

    test('matches snapshot', async () => {
        const { asFragment } = renderWithProviders(
            <MemoryRouter initialEntries={["/"]}>
                <App />
            </MemoryRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});