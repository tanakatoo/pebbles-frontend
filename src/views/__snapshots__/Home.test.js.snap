// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bg-primary-dark py-6 px-4"
    >
      <form
        action="#"
      >
        <div
          class="w-full flex  null gap-3"
        >
          <input
            class="rounded-ml grow placeholder-gray focus:ring-none"
            name="word"
            placeholder="検索"
            type="text"
            value=""
          />
        </div>
      </form>
    </div>
    <div
      class="flex justify-center bg-primary-super-light bg-homeHero md:bg-homeHeroTablet lg:bg-homeHeroDesktop bg-cover bg-center bg-no-repeat relative min-h-screen"
    >
      <div
        class="bg-gradient-to-b absolute from-primary-super-light from-0% via-transparent via-1% to-transparent to-100% w-full h-screen"
      />
      <h1
        class="text-primary-dark font-bold mx-4 text-center pt-10 md:pt-16 text-mobile-header-1 absolute"
      >
        英語の自己学習者、海外への旅行者＆任者をサポートするコミュニティー
      </h1>
    </div>
    <section
      class="pt-12 md:pt-24  mx-auto"
    >
      <h2
        class="text-center text-mobile-header-2 text-primary-dark font-medium"
      >
        Our mission 日本語
      </h2>
      <div
        class="mt-8 md:mt-12 lg:ps-8 lg:pe-8 flex flex-col justify-center items-center gap-8 lg:flex-row lg:w-full lg:space-around"
      >
        <div
          class="mb-5 flex flex-col lg:w-full"
        >
          <div
            class="mx-auto"
          >
            <img
              src="connect.svg"
            />
          </div>
          <p
            class="text-center mt-6 mx-12"
          >
            日本語 Help you connect with other expats and immigrants in Japan
          </p>
        </div>
        <div
          class="mb-5  flex flex-col lg:w-full"
        >
          <div
            class="mx-auto"
          >
            <img
              src="travel.svg"
            />
          </div>
          <p
            class="text-center mt-6 mx-12"
          >
            日本語 Help you get information for living and travelling in Japan
          </p>
        </div>
        <div
          class="mb-5  flex flex-col lg:w-full"
        >
          <div
            class="mx-auto"
          >
            <img
              class="w-[60px] h-[60px]"
              src="language.png"
            />
          </div>
          <p
            class="text-center mt-6 mx-12"
          >
            日本語 Help you find resources for studying Japanese
          </p>
        </div>
      </div>
    </section>
    <section
      class="pt-[84px] flex flex-col items-center"
    >
      <div
        class="w-full bg-homeHelp md:bg-homeHelpTablet lg:bg-homeHelpDesktop h-[160px] md:h-[200px] lg:h-[250px] bg-cover bg-center"
      />
      <div
        class="w-full text-center py-4 px-4 bg-primary text-white font-medium inline-flex items-center justify-center"
      >
        <p>
          日本語 Find connections. Share resources. Learn together.
        </p>
      </div>
    </section>
    <section
      class=""
    >
      <h2
        class=" py-16 md:py-24 px-4 text-mobile-header-2 font-medium text-center text-primary-dark"
      >
        コミュニティーをすべて探検しよう
      </h2>
      <div
        class=""
      >
        <div
          class="bg-background pt-8 pb-12 px-8 flex-col flex md:flex-row md:px-24 lg:px-48"
        >
          <div
            class="mx-auto shrink-0 p-4"
          >
            <img
              class="rounded-full object-cover w-48 h-48"
              src="studysupport-square.jpg"
            />
          </div>
          <div
            class="flex flex-col grow lg:ps-8"
          >
            <p
              class="text-center lg:text-left mt-6 mx-8 NotoSansJP font-medium text-mobile-page-header mb-4 text-primary-dark"
            >
              自己英語の学習サポート
            </p>
            <p
              class="text-black text-center lg:text-left lg:mx-8"
            >
              日本語 Learn pronunciation, listening, speaking, and vocabulary by just reading! We show, and support how to supplement and fast-forward your English learning for both kids and adults. Free trial available.
            </p>
            <div
              class="mx-auto lg:ms-8 mt-8"
            >
              <button
                class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-primary border-primary 
                text-white 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 
                hover:opacity-75"
                data-testid=""
                type="button"
              >
                <a
                  class="font-medium"
                  href="/study-support"
                >
                  Tell me more
                </a>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bg-study-buddy-accent pt-8 pb-12 px-8 flex-col flex md:flex-row md:px-24 lg:px-48"
        >
          <div
            class="mx-auto shrink-0 p-4"
          >
            <img
              class="rounded-full object-cover w-48 h-48"
              src="studyBuddy-square.jpg"
            />
          </div>
          <div
            class="flex flex-col grow lg:ps-8"
          >
            <p
              class="text-center lg:text-left mt-6 mx-8 font-StudyBuddyJA font-bold text-mobile-page-header mb-4 text-primary-dark"
            >
              スタディーバディー掲示板
            </p>
            <p
              class="text-black text-center lg:text-left lg:mx-8"
            >
              日本語 Become a study buddy and find like-minded students to study Japanese. Do a cultural and language exchange in the comfort of your home. Volunteer to teach Japanese people learn English.
            </p>
            <div
              class="mx-auto lg:ms-8 mt-8"
            >
              <button
                class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-primary border-primary 
                text-white 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 
                hover:opacity-75"
                data-testid=""
                type="button"
              >
                <a
                  class="font-medium"
                  href="/study-buddies"
                >
                  Tell me more
                </a>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bg-community-accent pt-8 pb-12 px-8 flex-col flex md:flex-row md:px-24 lg:px-48"
        >
          <div
            class="mx-auto shrink-0 p-4"
          >
            <img
              class="rounded-full object-cover w-48 h-48"
              src="community-square.jpg"
            />
          </div>
          <div
            class="flex flex-col grow lg:ps-8"
          >
            <p
              class="text-center lg:text-left mt-6 mx-8 font-Community font-medium text-gold-4 text-mobile-page-header mb-4 text-white"
            >
              英語街
            </p>
            <p
              class="text-white text-center lg:text-left lg:mx-8"
            >
              日本語 Join groups to practice writing and reading Japanese. Ask and answer any questions related to the Japanese language.
            </p>
            <div
              class="mx-auto lg:ms-8 mt-8"
            >
              <button
                class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-white 
                text-primary-dark 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 
                hover:opacity-75"
                data-testid=""
                type="button"
              >
                <span
                  class="font-medium"
                >
                  Tell me more
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bg-regional-accent pt-8 pb-12 px-8 flex-col flex md:flex-row md:px-24 lg:px-48"
        >
          <div
            class="mx-auto shrink-0 p-4"
          >
            <img
              class="rounded-full object-cover w-48 h-48"
              src="regional-square.jpg"
            />
          </div>
          <div
            class="flex flex-col grow lg:ps-8"
          >
            <p
              class="text-center lg:text-left mt-6 mx-8 font-Regional text-gold-4 text-mobile-page-header mb-4 text-white"
            >
              地域案内センター
            </p>
            <p
              class="text-white text-center lg:text-left lg:mx-8"
            >
              日本語 Get resources and connect with others wherever you are in the world. Ask and answer questions related to travelling and living in an area.
            </p>
            <div
              class="mx-auto lg:ms-8 mt-8"
            >
              <button
                class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-white 
                text-primary-dark 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 
                hover:opacity-75"
                data-testid=""
                type="button"
              >
                <span
                  class="font-medium"
                >
                  Tell me more
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bg-marketplace-accent pt-8 pb-12 px-8 flex-col flex md:flex-row md:px-24 lg:px-48"
        >
          <div
            class="mx-auto shrink-0 p-4"
          >
            <img
              class="rounded-full object-cover w-48 h-48"
              src="marketplace-square.jpg"
            />
          </div>
          <div
            class="flex flex-col grow lg:ps-8"
          >
            <p
              class="text-center lg:text-left mt-6 mx-8 font-EnglishMarketJA font-bold text-mobile-page-header mb-4 text-primary-dark"
            >
              マーケット
            </p>
            <p
              class="text-black text-center lg:text-left lg:mx-8"
            >
              日本語 Going back home? Moving to Japan? Buy, sell or offer your used items to future expats and immigrants.
            </p>
            <div
              class="mx-auto lg:ms-8 mt-8"
            >
              <button
                class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-primary border-primary 
                text-white 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 
                hover:opacity-75"
                data-testid=""
                type="button"
              >
                <span
                  class="font-medium"
                >
                  Tell me more
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="py-12"
    >
      <p
        class="py-8 px-4 text-mobile-page-header text-center text-primary-dark"
      >
        日本語　Whether you're a traveller, a student, an expat, immigrant or citizen,
      </p>
      <p
        class="px-4 text-mobile-page-header text-center text-primary-dark"
      >
        　日本語we are here to support you.
      </p>
      <div
        class="flex justify-center"
      >
        <button
          class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-primary border-primary 
                text-white 
                py-3 
                px-4 
                rounded-xl
                disabled:opacity-50 mt-12 w-[200px]
                hover:opacity-75"
          data-testid=""
          type="button"
        >
          <span
            class="font-medium"
          >
            More about us
          </span>
        </button>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
